File: Ahorn/entities/coreBerry.jl
module FrostHelperCoreBerry

using ..Ahorn, Maple

@mapdef Entity "FrostHelper/CoreBerry" FTCoreBerry(x::Integer, y::Integer, order::Integer=-1, checkpointID::Integer=-1, isIce::Bool=false)

const placements = Ahorn.PlacementDict(
	"Core Berry (Frost Helper)" => Ahorn.EntityPlacement(
		FTCoreBerry
	)
)

function Ahorn.selection(entity::FTCoreBerry)
	x, y = Ahorn.position(entity)

	return Ahorn.getSpriteRectangle("collectables/strawberry/normal00", x, y)
end

function Ahorn.renderAbs(ctx::Ahorn.Cairo.CairoContext, entity::FTCoreBerry, room::Maple.Room)
	x, y = Ahorn.position(entity)

	Ahorn.drawSprite(ctx, "collectables/strawberry/normal00", x, y)
end

end

File: Loenn/entities/coreBerry.lua
local coreBerry = {}

coreBerry.name = "FrostHelper/CoreBerry"
coreBerry.depth = -100
coreBerry.placements = {
    name = "normal",
    data = {
        order = -1,
        checkpointID = -1,
        isIce = false,
    }
}

function coreBerry.texture(room, entity)
    return entity.isIce and "collectables/FrostHelper/CoreBerry/Cold/ColdBerry_Cold00" or "collectables/FrostHelper/CoreBerry/Hot/CoreBerry_Hot00"
end

return coreBerry